lib/improver/nbhood/use_nbhood.py

```diff
@@ -173,7 +173,18 @@ def process(self, cube, mask_cube):
         xname = cube.coord(axis='x').name()
         result_slices = iris.cube.CubeList([])
         # Take 2D slices of the input cube for memory issues.
+        prev_x_y_slice = None
```

Comment:
Do we need a unit test for this functionality?

Reply:
Added

Reply:
I'm not sure it counts as a unit. If it were turned into a function, maybe.

Currently I don't see that if the added unit is doing (or isn't doing) what it is supposed to (namely skipping already calculated slices), that the test would fail or succeed accordingly.

Reply:
It is slightly unsatisfactory, I agree - I bunged in a print statement here to say 'Hey, I'm skipping this slice' to actually check the new unit test (and CLI test) was using this code - but then I obviously had to take it out.